cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppFabricHttpHandler.java

```diff
@@ -584,14 +584,8 @@ public void changeFlowletStreamConnection(HttpRequest request, HttpResponder res
   @Path("/apps/{app-id}")
   public BodyConsumer deploy(HttpRequest request, HttpResponder responder, @PathParam("app-id") final String appId,
                              @HeaderParam(ARCHIVE_NAME_HEADER) final String archiveName) {
-    try {
       return appLifecycleHttpHandler.deploy(rewriteRequest(request), responder, Constants.DEFAULT_NAMESPACE, appId,
```

Comment:
the method it delegates to already catches exception and returns null. This method will never receive any exception. it should simply delegate to v3 handler.